### ISSUE TITLE
API: recompute sequence lengths when updating an organism

### ISSUE THREAD

**Issue by abretaud** [GMOD/Apollo#1761]
I think we discussed it in montpellier, but I can't find a trace of it here and I don't remember if we found a solution for it. Feel free to close if it's irrelevant :)

When using the galaxy tool to update an organism, I'd like apollo to recompute all scaffolds length (and maybe other related things if there are) based on the new genome sequence.

This would be useful when upgrading a genome to a new assembly version. It could be a minor version with most of the genome identical except 1 or 2 nucleotide. Or even a major version with completely new sequence.

Apollo annotated genes should be kept for unchanged scaffolds, and removed for scaffolds that changed or disappeared (maybe make it optional?).

I guess this should be somewhere in the updateOrganism api

**Comment by erasche** [comment]
Yes, @moffmade and I have both experienced this where someone accidentally updates the sequence and it renders their genome inaccessible due to incorrect lengths.

**Comment by nathandunn** [comment]
I think this is an interesting idea, and I think I understand the use-case.    I guess I would assume that you would create a new organism for this use-case. 

Sure, we could recalculate the lengths for the underlying sequence.    You just need to call [loadRefSeqs](https://github.com/GMOD/Apollo/blob/master/grails-app/services/org/bbop/apollo/SequenceService.groovy#L250). 

I would just create a separate route for this.  You could put a flag in for this, as well. 

My question would be how you would shift any underlying annotations or sequence alterations?

**Comment by abretaud** [comment]
A separate route for this or adding a flag would be ok for me

For existing annotations or alterations, I don't really care about shifting them:
- if the scaffold/chromosome is unchanged, we leave them as is
- if the scaffold disappears -> we throw away annotations and alterations that were on it (galaxy tool could keep a backup I guess)
- and if the scaffold changed (like 10 more nucleotides) -> the user can choose to keep them as is (I'd assume the admin knows what he's doing) or throw them away

**Comment by nathandunn** [comment]
Hmm . .  so I'm trying to envision how this route might work and what level the route needs to be intelligent.    My suggestion is create a new route in OrganismController (or SequenceController) and annotate it and link it to a method *like* SequenceService.loadRefSeqs().  If you create a PR with this logic (and TODOs for areas you're unsure of) that gets it most of the way, I can finish it up (or help you finish it up).
